scipy/integrate/_ivp/base.py

```diff
@@ -100,6 +100,9 @@ class OdeSolver:
         Whether integration in a complex domain should be supported.
         Generally determined by a derived solver class capabilities.
         Default is False.
+    nstep : int, optional
```

Comment:
This is a bit tangential to this PR, but it is useful to support one main use case of the callback, which is to print progress.  Let me know if I should open a separate PR to include this in OdeSolver first.